engine/src/main/java/org/destinationsol/common/SolMath.java

```diff
@@ -67,146 +90,185 @@ public static float approach(float src, float dst, float speed) {
     }
 
     /**
-     * Builds a bound vector with the given angle and length.
+     * Builds a {@link Bound} vector with the given angle and length.
+     *
+     * @param angle Angle the vector should have
+     * @param len   Length the vector should have
      */
     @Bound
     public static Vector2 fromAl(float angle, float len) {
-        return fromAl(angle, len, false);
-    }
-
-    /**
-     * Builds a bound vector with the given angle and length. If not {@code precice}, the resulting vector angle may slightly differ from a given one, in the cost of performance.
-     */
-    @Bound
-    public static Vector2 fromAl(float angle, float len, boolean precise) {
         Vector2 vec = getVec();
-        fromAl(vec, angle, len, precise);
+        fromAl(vec, angle, len);
         return vec;
     }
 
     /**
-     * @return a new bound copy of src
+     * Builds a {@link Bound} vector set to the value of specified vector.
+     *
+     * @param src Vector to copy
+     * @return Bound copy of the vector
      */
     @Bound
     public static Vector2 getVec(Vector2 src) {
         return getVec(src.x, src.y);
     }
 
     /**
-     * @return a new bound vector
+     * Builds a {@link Bound} vector set to the specified value.
+     *
+     * @param x x value to set the vector to
+     * @param y y value to set the vector to
+     * @return Bound copy of the vector
      */
     @Bound
     public static Vector2 getVec(float x, float y) {
-        VECTORS_TAKEN++;
-        Vector2 v = vs.obtain();
+        vectorsTaken++;
+        Vector2 v = vectorPool.obtain();
         v.set(x, y);
         return v;
     }
 
     /**
-     * frees the bound vector. Don't use this vector after freeing!
+     * Frees the bound vector. The freed vector is meant to be no longer used after freeing.
+     *
+     * @param v Vector to free.
      */
     public static void free(Vector2 v) {
-        VECTORS_TAKEN--;
-        vs.free(v);
+        vectorsTaken--;
+        vectorPool.free(v);
     }
 
     /**
-     * @return a new bound vector
+     * Builds a {@link Bound} vector set to the value {@code (0, 0)}.
+     *
+     * @return Bound copy of the vector
      */
     @Bound
     public static Vector2 getVec() {
         return getVec(0, 0);
     }
 
     /**
-     * @return approximate cos of a degrees
-     */
-    public static float cos(float a) {
-        return MathUtils.cosDeg(a);
-    }
-
-    /**
-     * @return approximate sin of a degrees
+     * Check that an object has returned all the vectors he has borrowed.
+     *
+     * @param o Object to blame upon the not returning of vectors.
+     * @throws AssertionError When not all of the vectors were returned
      */
-    public static float sin(float a) {
-        return MathUtils.sinDeg(a);
+    public static void checkVectorsTaken(Object o) {
+        if (SolMath.vectorsTaken != 0) {
+            throw new AssertionError("vectors " + SolMath.vectorsTaken + ", blame on " + o);
```

Comment:
I'm not sure that my thinking is correct here or there's some odd convention I'm unaware of, but I've seen this pattern in the code of using `AssertionError` to outright crash the game over some (usually) relatively tiny issue rather than simply log an angry error message then continue. That looks really strange to me and I've taken it out a few places where I could make sense of some better form of error handling to harden the game.

I would suggest we try to avoid them unless somebody can illustrate why they're a good thing :-)

Reply:
This thing is here to avoid memory leaks of sort? If the vectors were not returned, they couldn't be used anymore, but nor would they be collected by garbage collector, since reference to them would still exist in the pool. Also, I was thinking about doing something similar to these vector pools to TS too, but there are several problems that can't be overcome easy enough, especially when the code hasn't been made with those... bound objects in mind. One of them is freeing the resources, as you need to be sure that no more refs to the object exist elsewhere before freeing them, and who knows what goes on in WorldGens and such?

Reply:
Also, error messages might go unnoticed, while crashes not. And even a tiny issue might grow into a big well-hidden bug if it goes unnoticed for a while

Reply:
Talked on Slack a bit, my comment was purely about the `AssertionError` causing a crash, the vector pool concept is definitely interesting :-)

I'd like to see us more toward using the CrashReporter and telemetry more to highlight errors, rather than risk full crashes over even small things. But yeah until we get there I don't mind leaving these in for better visibility. They do suck for hard to reproduce errors though (user keeps hitting mysterious crashes without obvious explanations) 

Will wait for @vampcat in any case on this PR since he commented about having pending stuff :)